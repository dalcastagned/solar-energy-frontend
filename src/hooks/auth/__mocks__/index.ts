export const userMock = {
  user: 'teste@teste.com',
  roles: [],
  token:
    'eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJBVVJPUkEiLCJhdWQiOiJBVVJPUkEiLCJleHAiOjMwMjc2NTMzNTAsInJvbGVzIjpbIlBBQ1NOR19ESUNPTV9CVUxLREFUQSIsIlBBQ1NOR19ESUNPTV9XQURPIiwiUEFDU05HX0RJQ09NX1FJRE8iLCJQQUNTTkdfQ09ORklHVVJBVElPTiIsIlBBQ1NOR19OT0RFX0NPTlRST0wiLCJQQUNTTkdfU0VSSUVTX01JR1JBVElPTiIsIlBBQ1NOR19TRVJJRVNfUkVNT1ZBTCJdLCJwcmVmZXRjaHJvbGVzIjpbXSwidXNlciI6InVzZXJ0ZXN0ZSJ9.8a_ZdpxfM1_rU-ekCaZnefGFsJVq2_LetC5yBD_80SE',
};

export const loginRequestMock = {
  email: 'usuario@usuario.com',
  password: 'password',
};

