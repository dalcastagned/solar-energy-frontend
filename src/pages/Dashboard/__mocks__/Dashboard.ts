export const counts = {
  activePlants: 0,
  inactivePlants: 0,
};

export const monthsGenerations = [
  {
    month: 'janeiro',
    generation: 150,
  },
];
