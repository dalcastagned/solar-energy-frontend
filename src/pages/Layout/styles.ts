import { Drawer, IconButton, ListItem, Avatar, MenuItem } from '@mui/material';
import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar';

import { styled } from '@mui/material/styles';

interface AppBarProps extends MuiAppBarProps {
  open?: boolean;
}

export const Main = styled('main')<{
  open?: boolean;
}>(({ theme, open }) => ({
  flexGrow: 1,
  padding: '2rem',
  transition: theme.transitions.create('margin', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  marginLeft: `-26rem`,
  '@media screen and (max-width: 680px)': {
    marginLeft: 0,
  },
  ...(open && {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  }),
}));

export const AppBar = styled(MuiAppBar)<AppBarProps>(({ theme, open }) => ({
  backgroundColor: '#fff',
  color: '#374557',
  fontWeight: 'bold',
  transition: theme.transitions.create(['margin', 'width'], {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  ...(open && {
    width: `calc(100% - 26rem)`,
    marginLeft: `26rem`,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  }),
}));

export const DrawerHeader = styled('div')(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: theme.spacing(0, 1),
  ...theme.mixins.toolbar,
  justifyContent: 'flex-end',
}));

export const IconButtonStyled = styled(IconButton)<{
  open?: boolean;
}>(({ open }) => ({
  mr: 2,
  '& .MuiSvgIcon-root': { fontSize: '2.25rem' },
  ...(open && { display: 'none' }),
}));

export const DrawerStyled = styled(Drawer)(() => ({
  width: '26rem',
  flexShrink: 0,
  '& .MuiDrawer-paper': {
    width: '26rem',
    boxSizing: 'border-box',
  },
  '& .MuiSvgIcon-root': { fontSize: '2.25rem' },
  '& .MuiTypography-root': { fontSize: '1.6rem' },

  '@media screen and (max-width: 680px)': {
    position: 'absolute',
  },
}));

export const Logo = styled('div')(() => ({
  display: 'flex',
  width: '100%',
  justifyContent: 'center',
  alignItems: 'center',
  paddingLeft: '3.85rem',

  '& img': {
    height: '5.5rem',
  },
}));

export const ItemList = styled(ListItem)(() => ({
  '& .MuiListItemButton-root': {
    borderRadius: '0.8rem',
  },
  '& .Mui-selected': {
    background: '#4CBC9A !important',
    color: '#fff',
    ':hover': { background: '#4CBC9AEE' },
    '& .MuiListItemIcon-root': { color: '#fff' },
  },
}));

export const ContainerAppBar = styled('div')(() => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  width: '100%',
}));

export const PaperStyles = {
  overflow: 'visible',
  filter: 'drop-shadow(0px 2px 8px rgba(0,0,0,0.32))',
  mt: 1.5,
  '& .MuiAvatar-root': {
    width: '4rem',
    height: '4rem',
    ml: -0.5,
    mr: 1,
  },
  '&:before': {
    content: '""',
    display: 'block',
    position: 'absolute',
    top: 0,
    right: 18,
    width: 10,
    height: 10,
    bgcolor: 'background.paper',
    transform: 'translateY(-50%) rotate(45deg)',
    zIndex: 0,
  },
};

export const AvatarStyled = styled(Avatar)(() => ({
  width: '4rem',
  height: '4rem',
  backgroundColor: '#374557',
  fontSize: '1.6rem',
}));

export const MenuItemStyled = styled(MenuItem)(() => ({
  fontSize: '1.6rem',
  '& .MuiSvgIcon-root': { fontSize: '2rem' },
}));
