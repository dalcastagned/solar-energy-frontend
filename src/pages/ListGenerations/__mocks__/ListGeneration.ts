export const generationsMock = {
  hasNextPag: false,
  hasPreviousPage: false,
  isFirstPage: true,
  isLastPage: true,
  pageCount: 1,
  pageNumber: 1,
  pageSize: 1,
  totalItemCount: 1,
  generations: [
    {
      id: 1,
      date: '2022-06-05T17:53:08.189Z',
      generatePower: 150,
    },
  ],
};

export const generationMock = {
  id: 1,
  date: '2022-06-05T17:53:08.189Z',
  generatePower: 150,
};

export const generationsWithPageMock = {
  hasNextPag: false,
  hasPreviousPage: false,
  isFirstPage: true,
  isLastPage: true,
  pageCount: 2,
  pageNumber: 1,
  pageSize: 1,
  totalItemCount: 1,
  generations: [
    {
      id: 1,
      date: '2022-06-05T17:53:08.189Z',
      generatePower: 150,
    },
  ],
};
