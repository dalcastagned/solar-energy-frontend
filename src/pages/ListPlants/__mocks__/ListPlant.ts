export const plantsMock = {
  hasNextPag: false,
  hasPreviousPage: false,
  isFirstPage: true,
  isLastPage: true,
  pageCount: 1,
  pageNumber: 1,
  pageSize: 1,
  totalItemCount: 1,
  plants: [
    {
      id: 1,
      nickname: 'nickname',
      place: 'place',
      brand: 'brand',
      model: 'model',
      active: true,
    },
  ],
};

export const plantMock = {
  id: 1,
  nickname: 'Apelido da Planta',
  place: 'Local da Planta',
  brand: 'Marca da Planta',
  model: 'Modelo da Planta',
  active: true,
};

export const plantsWithPageMock = {
  hasNextPag: false,
  hasPreviousPage: false,
  isFirstPage: true,
  isLastPage: true,
  pageCount: 2,
  pageNumber: 1,
  pageSize: 1,
  totalItemCount: 1,
  plants: [
    {
      id: 1,
      nickname: 'nickname2',
      place: 'place2',
      brand: 'brand2',
      model: 'model2',
      active: false,
    },
  ],
};
