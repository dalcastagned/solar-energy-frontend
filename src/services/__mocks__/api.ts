import axios from 'axios';

export type Error = {
  status?: number;
  data?: Record<string, unknown>;
  message?: string;
};

export type HttpResponse<DataType> = {
  data?: DataType;
  error?: Error;
};

const api = jest.genMockFromModule('axios') as jest.MockedFunction<
  typeof axios
>;

api.create = jest.fn(() => api);

export default api;
