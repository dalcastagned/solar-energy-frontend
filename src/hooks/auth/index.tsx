import {
  createContext,
  ReactNode,
  useCallback,
  useContext,
  useState,
} from 'react';

import jwt_decode, { JwtPayload } from 'jwt-decode';

import {
  LoginCredentials,
  UpdatePasswordCredentials,
  User,
} from '../../models/user';
import api from '../../services/api';

export type AuthState = {
  user: User;
};

export type AuthContextData = {
  user: User;
  signIn(credentials: LoginCredentials): Promise<void>;
  signOut(): void;
  updatePassword(credentials: UpdatePasswordCredentials): Promise<void>;
  signUp(credentials: LoginCredentials): Promise<void>;
};

export type AuthProviderParams = {
  children: ReactNode;
};

export const cleanStorage = (): void => {
  localStorage.removeItem('user');
};

export const handleToken = (token: string): void => {
  api.defaults.headers.common['X-JWT'] = token;
  api.defaults.headers.common.Authorization = `Bearer ${token}`;
};

const AuthContext = createContext<AuthContextData>({} as AuthContextData);

const AuthProvider = ({ children }: AuthProviderParams): JSX.Element => {
  const [data, setData] = useState<AuthState>(() => {
    const user = localStorage.getItem('user');
    const parsedUser = user ? JSON.parse(user) : null;
    const token = parsedUser?.token;

    if (token && user) {
      const tokenExpirationTime = jwt_decode<JwtPayload>(token).exp;

      const date = new Date().getTime() / 1000;

      if (tokenExpirationTime && tokenExpirationTime < date) {
        cleanStorage();

        return {} as AuthState;
      }

      handleToken(token);

      return { user: parsedUser };
    }
    return {} as AuthState;
  });

  const signIn = useCallback(async ({ email, password }: LoginCredentials) => {
    const response = await api.post<User>('/user/login', {
      email,
      password,
    });

    const { token } = response.data;
    const user = response.data;

    localStorage.setItem('user', JSON.stringify(user));

    handleToken(token);

    setData({ user });
  }, []);

  const updatePassword = useCallback(
    async ({
      user,
      currentPassword,
      newPassword,
    }: UpdatePasswordCredentials) => {
      await api.post<User>(`/user/reset-password/${user}`, {
        currentPassword,
        newPassword,
      });
    },
    [],
  );

  const signUp = useCallback(async ({ email, password }: LoginCredentials) => {
    await api.post<User>('/user/signup', {
      email,
      password,
    });
  }, []);

  const signOut = useCallback(() => {
    cleanStorage();

    setData({} as AuthState);
  }, []);

  return (
    <AuthContext.Provider
      value={{
        user: data.user,
        signIn,
        signOut,
        updatePassword,
        signUp,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = (): AuthContextData => useContext(AuthContext);

export { AuthProvider, useAuth };
