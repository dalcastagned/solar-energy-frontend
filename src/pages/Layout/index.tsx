import * as React from 'react';

import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import Logout from '@mui/icons-material/Logout';
import MenuIcon from '@mui/icons-material/Menu';
import PieChartIcon from '@mui/icons-material/PieChart';
import SettingsIcon from '@mui/icons-material/Settings';
import ShowChartIcon from '@mui/icons-material/ShowChart';
import { Menu } from '@mui/material';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import { useAuth } from 'hooks/auth';
import { useHistory } from 'react-router-dom';

import * as S from './styles';

type LayoutProps = {
  children?: React.ReactNode;
  Title?: string;
};

const Layout = ({ children, Title }: LayoutProps): JSX.Element => {
  const { signOut, user } = useAuth();
  const history = useHistory();
  const [open, setOpen] = React.useState(false);
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleClickAvatar = (event: React.MouseEvent<HTMLElement>): void => {
    setAnchorEl(event.currentTarget);
  };
  const handleCloseAvatar = (): void => {
    setAnchorEl(null);
  };

  const handleDrawerOpen = (): void => {
    setOpen(true);
  };

  const handleDrawerClose = (): void => {
    setOpen(false);
  };

  const handleLogout = (): void => {
    signOut();
    history.push('/');
  };

  const routes = [
    {
      id: '1',
      path: '/dashboard',
      title: 'Dashboard',
      icon: PieChartIcon,
    },
    {
      id: '2',
      path: '/plants',
      title: 'Unidade Consumidora',
      icon: ShowChartIcon,
    },
    {
      id: '3',
      path: '/generation',
      title: 'Cadastro de Energia Gerada',
      icon: SettingsIcon,
    },
  ];

  return (
    <Box sx={{ display: 'flex' }}>
      <CssBaseline />
      <S.AppBar position="fixed" open={open}>
        <Toolbar>
          <S.IconButtonStyled
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            open={open}
          >
            <MenuIcon />
          </S.IconButtonStyled>
          <S.ContainerAppBar>
            <Typography variant="h4" noWrap component="div">
              {Title}
            </Typography>
            <IconButton
              onClick={handleClickAvatar}
              size="small"
              sx={{ ml: 2 }}
              aria-controls={anchorEl ? 'account-menu' : undefined}
              aria-haspopup="true"
              aria-expanded={anchorEl ? 'true' : undefined}
            >
              <S.AvatarStyled>
                {user?.user.substring(0, 1).toUpperCase()}
              </S.AvatarStyled>
            </IconButton>
          </S.ContainerAppBar>
        </Toolbar>
      </S.AppBar>
      <Menu
        anchorEl={anchorEl}
        id="account-menu"
        open={Boolean(anchorEl)}
        onClose={handleCloseAvatar}
        onClick={handleCloseAvatar}
        PaperProps={{
          elevation: 0,
          sx: { ...S.PaperStyles },
        }}
        transformOrigin={{ horizontal: 'right', vertical: 'top' }}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
      >
        <S.MenuItemStyled sx={{ fontSize: '1.6rem' }} onClick={handleLogout}>
          <ListItemIcon>
            <Logout fontSize="small" />
          </ListItemIcon>
          Logout
        </S.MenuItemStyled>
      </Menu>
      <S.DrawerStyled variant="persistent" anchor="left" open={open}>
        <S.DrawerHeader>
          <S.Logo>
            <img src="/img/mini-logo.png" alt="Logo" />
          </S.Logo>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </S.DrawerHeader>
        <List sx={{ paddingTop: '2rem' }}>
          {routes.map(item => (
            <S.ItemList key={item.id}>
              <ListItemButton
                onClick={() => history.push(item.path)}
                selected={history.location.pathname.search(item.path) !== -1}
              >
                <ListItemIcon>
                  <item.icon />
                </ListItemIcon>
                <ListItemText primary={item.title} />
              </ListItemButton>
            </S.ItemList>
          ))}
        </List>
      </S.DrawerStyled>
      <S.Main open={open}>
        <S.DrawerHeader />
        {children}
      </S.Main>
    </Box>
  );
};

export default Layout;
