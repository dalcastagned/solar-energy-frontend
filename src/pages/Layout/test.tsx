import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { AuthProvider } from 'hooks/auth';
import { userMock } from 'hooks/auth/__mocks__';

import Layout from '.';
import { renderWithRoute } from '../../utils/tests/helpers';

describe('<Layout />', () => {
  it('should render the main', () => {
    const { container } = renderWithRoute(<Layout />);

    expect(container.firstChild).toMatchSnapshot();
  });
  it('should remove the user data from local storage when user sign out', async () => {
    localStorage.setItem('user', JSON.stringify(userMock));

    renderWithRoute(
      <AuthProvider>
        <Layout />
      </AuthProvider>,
    );

    const buttonProfile = screen.getByText('T');

    userEvent.click(buttonProfile);

    await waitFor(() => {
      const signOutButton = screen.getByText('Logout');

      userEvent.click(signOutButton);

      expect(localStorage.getItem('user')).toEqual(null);
    });
  });
  it('should remove the user data from local storage when user sign out', async () => {
    localStorage.setItem('user', JSON.stringify(userMock));

    renderWithRoute(
      <AuthProvider>
        <Layout />
      </AuthProvider>,
    );

    const buttonProfile = screen.getByText('T');

    userEvent.click(buttonProfile);

    await waitFor(() => {
      const signOutButton = screen.getByText('Logout');

      userEvent.click(signOutButton);

      expect(localStorage.getItem('user')).toEqual(null);
    });
  });
  it('should open and close Drawer', async () => {
    renderWithRoute(
      <AuthProvider>
        <Layout />
      </AuthProvider>,
    );

    userEvent.click(screen.getByTestId('MenuIcon'));

    expect(screen.getByTestId('ChevronLeftIcon')).toBeInTheDocument();

    userEvent.click(
      screen.getByRole('button', { name: 'Unidade Consumidora' }),
    );

    userEvent.click(screen.getByTestId('ChevronLeftIcon'));

    expect(screen.getByTestId('MenuIcon')).toBeInTheDocument();
  });
});
