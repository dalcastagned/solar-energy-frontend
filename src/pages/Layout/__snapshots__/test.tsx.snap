// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the main 1`] = `
<div
  class="MuiBox-root css-k008qs"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-u2d20r-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-u4wdza-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="css-nb2z2f"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap css-x00mp-MuiTypography-root"
        />
        <button
          aria-haspopup="true"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-9torc8-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-10818m2-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-u5rkgc-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
    >
      <div
        class="css-1k455el"
      >
        <div
          class="css-1ew8yep"
        >
          <img
            alt="Logo"
            src="/img/mini-logo.png"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-n81081-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-scufnu-MuiListItem-root"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PieChartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Dashboard
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-scufnu-MuiListItem-root"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShowChartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m3.5 18.49 6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Unidade Consumidora
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-scufnu-MuiListItem-root"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Cadastro de Energia Gerada
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>
  <main
    class="css-1brazwx"
  >
    <div
      class="css-1k455el"
    />
  </main>
</div>
`;
